main/FatesEcotypesMod.F90

```diff
@@ -0,0 +1,78 @@
+module FatesEcotypesMod
+
+  use FatesConstantsMod, only : r8 => fates_r8
+  use EDTypesMod, only : ed_site_type
+  use FatesPatchMod, only : fates_patch_type
+
+  implicit none
+  private  ! By default everything is private
+
+  ! Make public necessary subroutines and functions
+  public :: is_patch_forest
+
+contains
+
+  ! =====================================================================================
+
+  function does_patch_have_forest_tcthresh(patchptr, forest_tree_fraction_threshold)
+  ! DESCRIPTION:
+  ! Return boolean: Is this patch "forest"?
+  !
+  ! ARGUMENTS:
+  type(fates_patch_type), intent(in), pointer :: patchptr  ! pointer to patch object
+  real(r8), intent(in) :: forest_tree_fraction_threshold ! Tree fraction above which a patch is "forest"
+  !
+  ! RETURN VALUE
+  logical :: does_patch_have_forest_tcthresh
+  !
+  ! LOCAL VARIABLES
+  real(r8) :: tree_fraction = 0._r8
+
+  if (patchptr%area > 0._r8) then
+      tree_fraction = patchptr%total_tree_area / patchptr%area
+  else
+      tree_fraction = 0._r8
```

Comment:
This `tree_fraction = 0._r8` is unnecessary because it's initialized to 0 when it's defined.